test/cql-pytest/test_using_timestamp.py

```diff
@@ -90,3 +90,56 @@ def test_key_writetime(cql, table1):
         cql.execute(f'SELECT writetime(k) FROM {table1}')
     with pytest.raises(InvalidRequest, match='PRIMARY KEY part k'):
         cql.execute(f'SELECT ttl(k) FROM {table1}')
+
+# Reproducer for https://github.com/scylladb/scylladb/issues/14182
+def test_rewrite_using_same_timeout_and_ttl(scylla_only, cql, table1):
+    table = table1
+    k = unique_key_int()
+    ts1 = 1000
+    ttl1 = 10
+    ts2 = 1000
+    ttl2 = 10
+    values = [[1, 1], [1, 2], [2, 1]]
+    errors = []
+    for i in range(3):
+        v1, v2 = values[i]
+
+        t1 = time.time()
+        cql.execute(f"INSERT INTO {table} (k, v) VALUES ({k}, {v1}) USING TIMESTAMP {ts1} AND TTL {ttl1}")
+
+        # rewriting right away, so that both cells will carry the same expiry and ttl should return the larger value
+        rewrite = f"INSERT INTO {table} (k, v) VALUES ({k}, {v2}) USING TIMESTAMP {ts2} AND TTL {ttl2}"
+        cql.execute(rewrite)
+        select = f"SELECT * FROM {table} WHERE k = {k}"
+        res = list(cql.execute(select))
+        expected = v1 if v1 >= v2 else v2
+        if res[0].v != expected:
+            errors.append(f"Expected (k={k}, v={expected}) after immediate rewrite in iteration {i}, but got {res[0]}")
+
+        delay = 4
+        time.sleep(delay)
```

Comment:
Sad you need such a long sleep. To be honest, I don't understand why you need to sleep at all, can't you read the data and its ttl() immediately and see this problem without needing to wait for the TTL?
This also relates to my question about what it is that we are **really** fixing in this PR, I'll write it as a separate comment.

Reply:
I'd like to reproduce the actual issue, rather than testing for indirect evidence.
But I'll see if it can be reproduced reliably using short timeouts.

Reply:
Sometimes after you understand an issue you understand that although the customer noticed the problem only after the expiration, if they used "SELECT ttl(...)" they would have seen it immediately after the second write, so you don't need to wait for the expiration. But I guess I'm ok with this if a couple of 1-second sleeps are involved, and not a dozen 4-second sleeps.